double_linked_list/double_linked_list.c

```diff
@@ -0,0 +1,133 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include "double_linked_list.h"
+
+list* list_create(){ //how to tell if malloc worked?
+    list* l = (list*)malloc(sizeof(list));
+    l->quantity = 0;
+    l->head = NULL;
+    l->tail = NULL;
+    return l;
+}
+
+void list_free(list* l){
+    node* first = l->head;
+    while (first->next) {
+            node *second = first;
+            first = first->next;
+            free(second);
+    }
+    free(l);
+}
+
+void list_print(list* l){
+    if (!l)
+        printf("Empty list!\n");
+    else{
+        node* tmp = l->head;
+        printf("[ ");
+        while (tmp){
+            printf("%d ", tmp->data);
+            tmp = tmp->next;
+        }
+        printf("]\n");
+    }
+}
+
+bool list_prepend(list* l, int new_elem) {
+    node* new_node = (node*)malloc(sizeof(node));
+    if (!new_node)
+        return false;
+    new_node->data = new_elem;
+    new_node->next = l->head;
+    new_node->prev = NULL;
+    if (l->head)
+        l->head->prev = new_node;
+    l->head = new_node;
+    if (!l->head->next)
+        l->tail = l->head;
+    l->quantity += 1;
+    return true;
+}
+
+bool list_append(list* l, int new_elem){
+    node* new_node = (node*)malloc(sizeof(node));
+    if (!new_node)
+        return false;
+    new_node->data = new_elem;
+    new_node->prev = l->tail;
+    new_node->next = NULL;
+    if (!l->head)
+        l->head =  new_node;
+    else
+        l->tail->next = new_node;
+    l->tail = new_node;
+    l->quantity += 1;
+    return true;
+}
+
+int list_pop(list* l){
+    l->quantity -= 1;
+    node* tmp = l->tail;
+    int ret_el = tmp->data;
+    free(tmp);
+    if (l->quantity < 2)
+        l->tail = l->head;
```

Comment:
If there is only 1 element in the list, we will get double free (if we call `list_free` afterwards).